File: app/src/main/java/com/semanienterprise/softcom/models/SectionsItem.java
package com.semanienterprise.softcom.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class SectionsItem implements Serializable {

    @SerializedName("elements")
    private List<ElementsItem> elements;

    @SerializedName("label")
    private String label;

    public String getFormattedResults() {
        StringBuilder sb = new StringBuilder();
        sb.append("Results:\n");
        if (this.elements == null) return sb.toString();
        for (ElementsItem element : this.elements) {
            sb.append(element.getFormattedResult() + "\n");
        }
        return sb.toString();
    }

    @Override
    public String toString() {
        return
                "SectionsItem{" +
                        "elements = '" + elements + '\'' +
                        ",label = '" + label + '\'' +
                        "}";
    }
}

File: app/src/main/java/com/semanienterprise/softcom/models/SectionsItem.kt
package com.semanienterprise.softcom.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class SectionsItem : Serializable {

    @SerializedName("elements")
    var elements: List<ElementsItem>? = null
        set(elements) {
            field = this.elements
        }

    @SerializedName("label")
    var label: String? = null
        set(label) {
            field = this.label
        }

    val formattedResults: String
        get() {
            val sb = StringBuilder()
            sb.append("Results:\n")
            if (this.elements == null) return sb.toString()
            for (element in this.elements!!) {
                sb.append(element.formattedResult + "\n")
            }
            return sb.toString()
        }

    override fun toString(): String {
        return "SectionsItem{" +
                "elements = '" + this.elements + '\''.toString() +
                ",label = '" + this.label + '\''.toString() +
                "}"
    }
}